app/(main)/league/all/page.test.tsx

```diff
@@ -1,124 +1,295 @@
+  });
+
+  test('should handle form submission to join a league', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    const user = {
+      documentId: '123',
+      email: 'test@test.com',
+      id: '123',
+      leagues: ['12'],
+    };
+
+    const league = {
+      leagueId: '123',
+      leagueName: 'Test League',
+      logo: 'logo.png',
+      participants: [],
+      survivors: [],
+    };
+
+    const updateUser = jest.fn();
+
+    mockUseDataStore.mockReturnValue({
+      user,
+      allLeagues: [league],
+      updateUser,
+    });
+
+    mockGetUserLeagues.mockResolvedValueOnce([]);
+    mockGetAllLeagues.mockResolvedValueOnce([league]);
+    mockAddUserToLeague.mockResolvedValue(
+      Promise.resolve({
+        documentId: user.documentId,
+        leagueId: league.leagueId,
+        selectedLeagues: [...(user.leagues ?? []), league.leagueId],
+        participants: [...(league.participants ?? []), user.id],
+        survivors: [...(league.survivors ?? []), user.id],
+      }),
+    );
 
     render(<Leagues />);
 
     await waitFor(() => {
       expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
     });
 
-    const selectElement = screen.getByLabelText(/Select league to join/i);
-    expect(selectElement).toBeInTheDocument();
+    const selectElement = screen.getByTestId('select-available-leagues');
+    fireEvent.change(selectElement, { target: { value: '123' } });
+    fireEvent.click(screen.getByTestId('join-league-button'));
+
+    await waitFor(() => {
+      expect(mockAddUserToLeague).toHaveBeenCalledWith({
+        userDocumentId: user.documentId,
+        selectedLeague: league.leagueId,
+        selectedLeagues: [...(user.leagues ?? []), league.leagueId],
+        participants: [...(league.participants ?? []), user.id],
+        survivors: [...(league.survivors ?? []), user.id],
+      });
+      expect(updateUser).toHaveBeenCalledWith(
+        user.documentId,
+        user.id,
+        user.email,
+        [...user.leagues, league.leagueId],
+      );
+      expect(toast.custom).toHaveBeenCalledWith(
+        <Alert
+          variant={AlertVariants.Success}
+          message={`Added ${league.leagueName} to your leagues!`}
+        />,
+      );
+    });
+  });
+
+  test('should show error if adding to league fails', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    const user = {
+      documentId: '123',
+      email: 'test@test.com',
+      id: '123',
+      leagues: ['12'],
+    };
 
+    const league = {
+      leagueId: '123',
+      leagueName: 'Test League',
+      logo: 'logo.png',
+      participants: [],
+      survivors: [],
+    };
+
+    mockUseDataStore.mockReturnValue({
+      user,
+      allLeagues: [league],
+    });
+
+    mockGetUserLeagues.mockResolvedValueOnce([]);
+    mockGetAllLeagues.mockResolvedValueOnce([league]);
+    mockAddUserToLeague.mockResolvedValue(
+      Promise.resolve({
+        documentId: user.documentId,
```

Comment:
same comment as above:

```
  userDocumentId: user.documentId,
        selectedLeague: league.leagueId,
        selectedLeagues: [league.leagueId],
        participants: [user.id],
        survivors: [user.id],
        ```

Reply:
The reason why I set up this code like this is to match the code in league/page.tsx file. What are your thoughts on the different formats?

Reply:
@vmaineng it's not about different formats. It's about what they are doing.

In page.tsx, the code handles adding a new participant/survivor as follows:

- It spreads the existing participants/survivors (if any).
- It then adds the new participant/survivor to this spread array.
- If there's no existing history of participants/survivors, it starts with an empty array before adding the new user.

The test is verifying that the function is called with the correct data you're passing into it. By having it the same way as page.tsx you are duplicating code.

Reply:
Updated :)